### DISCUSSION TITLE
export environment before clone?

### DISCUSSION THREAD

**Discussion by xenoterracide** [z-shell/zi#132] [Q&A] [answered]
I've noticed that our corporate proxy is blocking some stuff on github and that they have `wait` and aren't using `for` I'm not sure I understand the difference... for example, and this is called in a function, for which I tried calling our proxy connection script, but it doesn't seem to be making them available here.

```zsh
  zi ice lucid wait cloneonly proto"git" nocompile notify \
    atclone"install --compare -D colors/desertink.vim $VIM_COLOR/desertink.vim" \
    atpull"%atclone"
  zi light toupeira/vim-desertink
```

for some reason it's only these

```zsh
# ~/.zshrc
function vim_setup() {
   cofproxy on # I connect to the corporate proxy

  local VIMDIR="$HOME/.vim"
  local VIM_COLOR="$VIMDIR/colors"
  local VIM_PACK="$VIMDIR/pack"
  local VIM_FTPLUGIN="$VIMDIR/ftplugin"
  local VIM_FTDETECT="$VIMDIR/ftdetect"
  local VIM_AFTER_SYNTAX="$VIMDIR/after/syntax"


  zi ice lucid wait cloneonly proto"git" nocompile notify \
    atclone"install --compare -D colors/desertink.vim $VIM_COLOR/desertink.vim" \
    atpull"%atclone"
  zi light toupeira/vim-desertink

  zi ice lucid wait cloneonly proto"git" nocompile \
    atclone"install -D after/syntax/zsh.vim $VIM_AFTER_SYNTAX/zsh.vim" \
    atpull"%atclone"
  zi light z-shell/zinit-vim-syntax

  zi ice lucid wait cloneonly proto"git" nocompile \
    atclone"install -D syntax/systemd.vim $VIM_AFTER_SYNTAX/systemd.vim; install -D ftplugin/systemd.vim $VIM_FTPLUGIN/systemd.vim; install -D ftdetect/systemd.vim $VIM_FTDETECT/systemd.vim" \
    atpull"%atclone"
  zi light wgwoods/vim-systemd-syntax

#  zi ice lucid wait cloneonly proto"git" nocompile \
#    atclone"rsync --mkpath --exclude '.*' --archive . $VIM_PACK/local/start/editorconfig" \
#    atpull"%atclone"
#  zi load editorconfig/editorconfig-vim
}
vim_setup
```

```zsh
# ~/.zshenv

if [ -f /usr/local/cof/bin/cofproxy ]; then 
  source /usr/local/cof/bin/cofproxy > /dev/null 2>&1
  cofproxy on
fi
```

this exists outside fo the function and has no issue whatsoever
```zsh
zi wait for \
  OMZL::theme-and-appearance.zsh atload"unalias ls" \
  OMZP::colored-man-pages \
  OMZP::systemd \
  OMZP::direnv \
  OMZP::gitignore
  
```

**Comment by xenoterracide** [comment, marked as answer]
Fixed it, the problem was actually that I had set `proto"git"` which is blocked

**Reply by ss-o** [reply]
> I've noticed that our corporate proxy is blocking some stuff on github and that they have wait and aren't using for I'm not sure I understand the difference... for example, and this is called in a function, for which I tried calling our proxy connection script, but it doesn't seem to be making them available here.

@xenoterracide, if you read the docs and there is something unclear, this means I have to improve the docs. Let me know in such a case. Thanks. :+1:

**Reply by xenoterracide** [reply]
Actually, the most unclear thing when something is blocked or otherwise unavailable such as a bad URL... The errors are terrible. It's hard to tell what it's reaching out to and what protocol it's using.

**Reply by ss-o** [reply]
More verbose calls? Could you open an issue for this?
To be specific as possible on the exact moments it should appear.

Also if you would have your "best case scenario" in other words the way you prefer things to happen. So can set a target/guideline.

Additionally what about adding "https://github/cli/cli" internally, under ZI control?

As I am currently working in relation to this so all suggestions are very helpful, Thanks,

**Reply by ss-o** [reply]
After the last merge, failed ice functions will return a status code instead of silence.
It can be observed in logs: https://github.com/z-shell/zd/actions/runs/2240394915
